public/app/core/components/TagFilter/TagFilter.tsx

```diff
@@ -45,10 +52,24 @@ export const TagFilter: FC<Props> = ({
   tags,
   width,
 }) => {
-  const theme = useTheme();
-  const styles = getStyles(theme);
+  const styles = useStyles2(getStyles);
+
+  const currentlySelectedTags = tags.map((tag) => ({ value: tag, label: tag, count: 0 }));
+  const [options, setOptions] = useState<TagSelectOption[]>(currentlySelectedTags);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectKey, setSelectKey] = useState<string>(JSON.stringify(tags));
+  const [previousTags, setPreviousTags] = useState(tags);
 
-  const onLoadOptions = async (query: string) => {
+  useEffect(() => {
+    if (updateOptionsDynamically && previousTags !== tags) {
+      setPreviousTags(tags);
+      // Slight delay necessary to ensure query results are most up to date
+      // Key switching approach seems to be best way to ensure consistent update of tag options
+      setTimeout(() => setSelectKey(JSON.stringify(tags)), 100);
```

Comment:
Seems like tag options don't dynamically update after selection unless we continue to do this key switching (in effect forcing React to re-render the select component and load up to date options)

I can continue looking into alternative approaches as this is not an ideal solution